### PR TITLE
Add manual attendence

### PR DESCRIPTION
**Pull request by SaswatB**
For issue #485

### CODE REVIEW

File: client/app/admin/admin.html [hunk 1]
```diff
@@ -45,6 +45,7 @@
             <td colspan="2">Small Group</td>
             <td colspan="2">Contributions</td>
             <td>Toggle</td>
```
Comment (kmcnellis) [inline client/app/admin/admin.html]: You should change `<td>Toggle</td>` to `<td colspan="2">Edit</td>`, then move  `<td>Attendance</td>` to the second header line after  `<td sort-arrow sortby='active' text='Active' sortorder='sortorder'></td>`.

File: client/app/admin/admin.html [hunk 2]
```diff
@@ -80,7 +81,11 @@
                 <a class="btn btn-link md-text no-padding" role="button" ng-href="#" ><i class="fa"  ng-class="{'fa-minus-square-o':!user.active, 'fa-check-square-o':user.active}" ></i></a>
                 <!-- <input class="center-link" type="checkbox" ng-model="user.activeChange"></input> -->
               </td>
-
+              <td class="no-padding">
+                <a class="btn btn-link large-text no-padding" role="button" ng-click="addAttendence(user)" title="Add Attendance">
+                  <i class="fa fa-calendar-plus-o" ></i>
```
Comment (kmcnellis) [inline client/app/admin/admin.html]: The calendar icon is a bit too big.

File: client/components/addAttendance/addAttendance.directive.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        $scope.attend.type = "Bonus Day";
+    }
+  }
+
+  $scope.submit = function(form) {
+    if($scope.submitted) {
+      return;
+    }
+    if(!form.$valid) {
+      notify("Please fill out all fields before submitting");
```
Comment (kmcnellis) [inline client/components/addAttendance/addAttendance.directive.js]: This should use angular form validation, not a floating message

File: client/components/addAttendance/addAttendance.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  <div class="modal-body">
+    <form class="form" name="addAttendanceForm" ng-submit="submit(addAttendanceForm)" novalidate>
+
+      <div>
+        <label>Attendance Date</label>
+        <div uib-datepicker class="well well-sm" ng-model="attend.date" ng-change="autofill()" required></div>
+      </div>
+      <div class="form-group">
+        <label for="attendanceType">Attendance Type</label>
+        <select class="form-control" id="attendanceType" ng-model="attend.type" required>
```
Comment (kmcnellis) [inline client/components/addAttendance/addAttendance.html]: Here is where you would add a form validation error message

File: server/api/attendance/attendance.controller.js [hunk 5]
```diff
@@ -613,6 +613,45 @@ exports.setAttendanceFullBonus = function(req, res) {
...
+exports.addManualAttendance = function(req, res) {
+  var userId = req.params.user;
+  var date = req.body.date;
+  var smallgroup = req.body.smallgroup
+  var bonusDay = req.body.bonusday;console.log("1")
+
+  return User.findById(userId, function(err,user){
+    if (err) {
+      return handleError(err);
+    }console.log("2")
```
Comment (kmcnellis) [inline server/api/attendance/attendance.controller.js]: Remove console.log

File: server/api/attendance/attendance.controller.js [hunk 6]
```diff
@@ -613,6 +613,45 @@ exports.setAttendanceFullBonus = function(req, res) {
...
+    return ClassYear.getCurrent(function(err, classYear){
+      if (err) {
+        return handleError(err);
+      }console.log("3")
```
Comment (kmcnellis) [inline server/api/attendance/attendance.controller.js]: Remove console.log

File: server/api/attendance/attendance.controller.js [hunk 7]
```diff
@@ -613,6 +613,45 @@ exports.setAttendanceFullBonus = function(req, res) {
...
+
```
Comment (kmcnellis) [inline server/api/attendance/attendance.controller.js]: You should first check that there isn't already an attendance of that type for the user submitted on that day.

File: client/components/addAttendance/addAttendance.directive.js [hunk 8]
```diff
@@ -15,16 +17,26 @@ angular.module('observatory3App')
...
+    $scope.formTypeEmptyError = !$scope.attend.type;
   }
 
   $scope.submit = function(form) {
     if($scope.submitted) {
       return;
     }
     if(!form.$valid) {
-      notify("Please fill out all fields before submitting");
+      console.log(!!$scope.attend.date);
```
Comment (kmcnellis) [inline client/components/addAttendance/addAttendance.directive.js]: console.log again

**Comment by agundy** [comment]
@kmcnellis and @SaswatB what is the status of this? Looks like the last update was a week ago

**Comment by kmcnellis** [comment]
Looks good to me. @agundy you were reviewing the changes?

**Pull request merged**